lib/publiccloud/noprovider.pm

```diff
@@ -0,0 +1,58 @@
+# SUSE's openQA tests
+#
+# Copyright 2018 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Helper class for dummy provider
+#
+# Maintainer: qa-c team <qa-c@suse.de>
+
+package publiccloud::noprovider;
+use Mojo::Base 'publiccloud::provider';
+use testapi;
+use publiccloud::existing;
+
+has ssh_key => undef;
+has ssh_key_file => undef;
+
+sub init {
+    my ($self) = @_;
+    $self->SUPER::init();
+    $self->provider_client(publiccloud::existing->new());
+    $self->provider_client->init();
+}
+
+sub create_instances {
+    my ($self, %args) = @_;
+    $args{check_connectivity} //= 1;
+
+    my @vms;
+    my $instance = publiccloud::instance->new(
+        public_ip => get_var('PUBLIC_CLOUD_EXISTING_ADDRESS'),
+        username => 'ec2-user',
+        ssh_key => '~/.ssh/id_rsa',
+        provider => $self
+    );
+    # Install server's ssh publicckeys to prevent authenticity interactions
+    assert_script_run(sprintf('ssh-keyscan %s >> ~/.ssh/known_hosts', $instance->public_ip));
+    push @vms, $instance;
+    return @vms;
+}
+
+sub create_ssh_key {
+    my ($self, %args) = @_;
+    $args{ssh_private_key_file} //= '/root/.ssh/id_rsa';
+    assert_script_run('SSH_DIR=`dirname ' . $args{ssh_private_key_file} . '`; mkdir -p $SSH_DIR');
+
+    # ssh user@instance awk 'NF {sub(/\r/, ""); printf "%s\\n",$0;}' ~/.ssh/id_rsa; echo
```

Comment:
leftovers ?

Reply:
No 😄 It's documentation how to generate the ssh key